### PR TITLE
Extending Sumatra with plugins.

### PR DESCRIPTION
**Pull request by apdavison**

### CODE REVIEW

File: doc/plugins.txt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    $ smt configure --remove-plugin=php
+
+
+Component interfaces
+====================
+
+As shown in the PHP example above, writing a custom component involves writing
+a new class which inherits from a Sumatra base class, and then registering the
+class with the component registry.
```
Comment (timtroendle) [inline doc/plugins.txt]: The component registry is not documented if I'm not mistaking; maybe that would be the chance to do so.

While checking it, I thought about the following: if I'm getting this right, a plugin should never call `add_component_type` but only `register` on the registry. To hide that entirely, the `registry` class and unique instance could be hidden using a `_` syntax (by the way, shouldn't this be implemented as a singleton as in `test_projects.py`?) Instead, the registration could happen through a decorator at the class. That would have the additional benefit of wrapping the whole singleton handling as well. What do you think?

File: doc/plugins.txt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    registry.register(PHPcli)
+
+To use this plug-in in a Sumatra project, it should be saved in a file (e.g. :file:`php.py`)
+that is on the Python module search path (for example, installed in :file:`site-packages` or
+added to the :envvar:`PYTHONPATH` environment variable), and then added to the project using:
+
+.. code-block:: bash
+
+    $ smt configure --add-plugin=php
```
Comment (timtroendle) [inline doc/plugins.txt]: Would be nice if I could configure that globally for myself or the whole machine (that could be as in Git using `--global` or `--system`). If I'd have my own recordstore implementation for example, I most probably would like to have that defined globally.

From `users.py` I know that there have been ideas of a per-user configuration already before. What do you think of extending that for plugins?

Comment (apdavison) [inline doc/plugins.txt]: Yes, a global configuration is a good idea. I don't want to do it as part of this PR, so could you create a ticket for this?

File: doc/plugins.txt (hunk 1 above)
Comment (apdavison) [inline doc/plugins.txt]: I like the idea of a register decorator, although I don't have time to implement it at the moment. Maybe one for a future PR?

File: doc/plugins.txt (hunk 2 above)
Comment (timtroendle) [inline doc/plugins.txt]: Sure, I created #247 covering that.

File: doc/plugins.txt (hunk 1 above)
Comment (timtroendle) [inline doc/plugins.txt]: Sure. I've created a sketch for this already. I can bring this up after this PR is merged.

**Pull request merged**